### PR TITLE
[IE] Add new API to return the number of logical cores

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - *Create new API to return the number of logical cores for BIG and All cores*
 - *This is needed for gpu plugin's cpu affinity control*

### Tickets:
 - *63585*

### CODE REVIEW

**Comment by ahnyoung-paul** [comment]
@myshevts @vladimir-paramuzov @ilya-lavrenov Could you please review this PR?

**Comment by yeonbok** [comment]
@myshevts  We have other PR which has dependency to this PR. Could you please review this? Thanks!

**Review by myshevts — APPROVED** [review]

File: inference-engine/src/plugin_api/ie_system_conf.h [hunk 1]
```diff
@@ -52,6 +52,16 @@ INFERENCE_ENGINE_API_CPP(std::vector<int>) getAvailableCoresTypes();
...
 
+/**
+ * @brief      Returns number of CPU logical cores on Linux/Windows (on other OSes it simply relies on the original
+ * parallel API of choice, which uses the 'all' logical cores). call function with 'false' to get #logical cores of
+ * all types call function with 'true' to get #logical 'Big' cores number of 'Little' = 'all' - 'Big'
+ * @ingroup    ie_dev_api_system_conf
+ * @param[in]  bigCoresOnly Additionally limits the number of reported cores to the 'Big' cores only.
+ * @return     Number of logical CPU cores.
+ */
+INFERENCE_ENGINE_API_CPP(int) getNumberOfLogicalCPUCores(bool bigCoresOnly = false);
```
Comment (ilya-lavrenov) [inline inference-engine/src/plugin_api/ie_system_conf.h:63]: can we add some tests?

Comment (yeonbok) [reply inference-engine/src/plugin_api/ie_system_conf.h:63]: Found that the original "getNumberOfCPUCores" is tested by task_executor_tests.cpp. @ahnyoung-paul  could you add the new APIs' tests there too?

Comment (ahnyoung-paul) [reply inference-engine/src/plugin_api/ie_system_conf.h:63]: Okay, I see. I will add API's test case. thanks.

Comment (ahnyoung-paul) [reply inference-engine/src/plugin_api/ie_system_conf.h:63]: I add tests using getNumberOfLogicalCPUCores. Please review it again.

**Review by yeonbok — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/src/plugin_api/ie_system_conf.h (hunk 1 above)
Comment (ilya-lavrenov) [reply inference-engine/src/plugin_api/ie_system_conf.h:63]: thanks!

**Review by apankratovantonp — APPROVED** [review]

**Pull request merged**